test/Microsoft.AspNetCore.Mvc.ViewFeatures.Test/Rendering/HtmlHelperSelectTest.cs

```diff
@@ -1299,6 +1301,30 @@ public void GetEnumSelectListTEnum_ThrowsWithNonEnum()
         }
 
         [Fact]
+        public void GetEnumSelectListTEnum_LocalizedEnum()
+        {
+            // Arrange
+            var stringLocalizer = new Mock<IStringLocalizer>();
+            stringLocalizer
+                .Setup(s => s[It.IsAny<string>()])
+                .Returns<string>((s) => { return new LocalizedString(s, s + " " + CultureInfo.CurrentCulture); });
+            var stringLocalizerFactory = new Mock<IStringLocalizerFactory>();
+            stringLocalizerFactory
+                .Setup(s => s.Create(It.IsAny<Type>()))
+                .Returns(stringLocalizer.Object);
+
+            var metadataProvider = TestModelMetadataProvider.CreateDefaultProvider(stringLocalizerFactory.Object);
+            var metadata = metadataProvider.GetMetadataForType(typeof(EnumWithFields));
+            var htmlHelper = new TestHtmlHelper(metadataProvider);
+
+            // Act
+            var result = htmlHelper.GetEnumSelectList<EnumWithDisplayNames>();
+
+            // Assert
+            Assert.Equal("cero en-US", result.Where(s => s.Value == "0").First().Text);
```

Comment:
Do the same `Assert.All()` thing I mentioned [earlier](https://github.com/aspnet/Mvc/pull/5185/files#r77050742).